### DISCUSSION TITLE
RFC #2 - Immediate Concerns

### DISCUSSION THREAD

**Discussion by RaddishIoW** [HaylLtd/cordova-plugin-background-geolocation#8] [General]
## Proposal
It is proposed to create a release as soon as possible, with the most pressing concerns and issues resolved in order to secure the name on NPM and provide the community with a working plugin for background geolocation.
A milestone should be created on GitHub to organise this release and the issues that should be resolved in it.
The version number will be 1.0.0, as this is the first release of this plugin on NPM.

**Comment by HarelM** [comment]
I agree this version should be an in place replacement to original fork to allow people who are using this version to migrate without any breaking changes in terms of API.
Having said that, I think (and it's open to discussion) that the complexity of the submodules should be addressed in this version (to reduce complexity and it can be in this version as the API and everything else should not be altered).

**Reply by RaddishIoW** [reply]
Agreed, I think we should move the submodules into the main repository. I'll make an issue for this.

**Comment by RaddishIoW** [comment]
I'm going to attempt to get the submodules merged into the main repository this coming week, and then I think we should get a version published to NPM.
I've merged the PR that fixes the debug crash on iOS... is there anything else anyone thinks should be in this first release?

**Reply by HarelM** [reply]
Great! Let's have the first version in npm so I can test that it works and I can replace it in my app and continue to our next versions...

**Comment by RaddishIoW** [comment]
@HarelM I think we're ready for another RC? Shall I kick this one off? - would be good for me to know how to do it, as I haven't done one yet!

I'll test this one this eve on iOS and Android.

**Reply by HarelM** [reply]
Yes. please kick-off one.
It's very simple - commit the version bump and create a release in github with the relevant version name.

**Reply by RaddishIoW** [reply]
Hmmm... Doesn't appear to have triggered the workflow to run for some reason. At first glance, I can't see what could be missing...

**Reply by HarelM** [reply]
weird... how did you create the release? from github UI?

**Reply by RaddishIoW** [reply]
Yep - "create a Release" on the right underneath the Tags. Had a look at the workflow and from what I can see it should have just triggered, although I haven't personally used the on: release trigger before.

**Reply by RaddishIoW** [reply]
Thinking about it, the only other thing that might have affected it is that I created the release as a draft first - I'd forgotten to bump the version, so created the draft release, bumped and committed, then published the release.

**Reply by HarelM** [reply]
Ahh, it might explain.
I've delete the tag, release and created them again and it worked.
Let's make sure you run the next one to make sure it works for you.
You can also create rc.6 to make sure it works for you, it's not like we care much about the numbering right now...

**Comment by HarelM** [comment]
Is there anything else we would like to do before releasing version 1.0.0 (except maybe testing it more)?

**Reply by RaddishIoW** [reply]
I don't think so - there's nothing I've come up against yet that needs sorting. I'd say let's get a full v1.0.0 published and if there are any issues found we can patch later.

We should treat the `master` branch as the 'stable' or 'production' branch, so when we create a stable release from the develop branch, we should merge those changes into that branch on release. Sound good? We could automate this at some point in the future maybe... are you familiar with [gitflow](https://blog.axosoft.com/gitflow/)?

**Reply by RaddishIoW** [reply]
Hmm... May have spoken too soon! My android build is crashing on launch, but this might be stale things from a previous build.
Have you seen any issues with Android builds crashing?

**Reply by RaddishIoW** [reply]
This is the error I'm getting: `java.lang.RuntimeException: Unable to get provider com.marianhello.bgloc.data.provider.LocationContentProvider: android.content.res.Resources$NotFoundException: String resource ID #0x0`

**Reply by RaddishIoW** [reply]
Ignore me - my install had outdated stuff. I had to remove and re-add the plugin to upgrade to the latest version.

**Reply by HarelM** [reply]
Would you like to do the honor of creating the first non pre-release version?
Don't forget to create the release after you commit the version bump :-)

**Reply by RaddishIoW** [reply]
OK yeah, will do - as soon as I can get the RC working in my app! I'm having all kinds of issues with it for some reason. I think there's still a reference to the old `cordova-plugin-*` plugin somewhere in my cordova metadata, but I can't find out where at the moment.
I'm getting the following in my console on running:
`plugins/cordova-plugin-background-geolocation/www/BackgroundGeoLocation.js:1 Failed to load resource: net::ERR_FILE_NOT_FOUND`
and
```
Uncaught Error: Module cordova-plugin-background-geolocation.BackgroundGeoLocation does not exist.
    at addEntry (cordova.js:1327)
    at Object.exports.clobbers (cordova.js:1337)
    at onScriptLoadingComplete (cordova.js:1646)
    at scriptLoadedCallback (cordova.js:1679)
    at HTMLScriptElement.<anonymous> (cordova.js:1633)
```
I've searched my whole project for references to the old plugin, and caught some in the `package.json` and elsewhere, but I'm still getting this... Any ideas? I expect I'm missing something obvious.

**Reply by HarelM** [reply]
Delete the platform folder and the plugins folder and simply add android platform back again.
If this doesn't work I'm not sure what will.
I have a CI build which always creates the app from scratch so I always know if this is my environment issue or a real one...

**Reply by HarelM** [reply]
I've also used the command to add the plugin with the specific version i.e. `cordova plugin add c-b-g-plugin@1.0.0-rc#` it might also help, I don't know...
I needed to remove some manual stuff from the package.json file due to bad configuration related to the old plugin.
You can check my changes here:
https://github.com/IsraelHikingMap/Site/compare/1427-gps-transaction

**Reply by RaddishIoW** [reply]
I just ran `cordova plugins list` and found `cordova-plugin-background-location v1.0.6` listed! No idea where that came from, I haven't installed that, but it must have come from somewhere - but I've deleted it and will clear out the folders as you suggest and try again.

**Reply by RaddishIoW** [reply]
OK, got it running - thanks for the tips :)
I'll publish the full version now then!

**Comment by HarelM** [comment]
We need to see how to get the tests up and running... or at least understand if we want to run them...

**Reply by RaddishIoW** [reply]
Yeah, tests are essential really I think.
Maybe the tests should be part of the example? Then we'd have a framework for the plug-in to run in for proper integration tests.
Unit tests could be useful for testing individual functions and things... I think there are tests for the Android and iOS code already, but it sure how to run them at the moment. That's why I left the CI stuff in the repo for the moment, as we could look at how they were run before...

**Reply by HarelM** [reply]
We probably need to understand what's currently available and then decide how to improve it. I'll open an issue to track it.
